bluez-async/examples/ruuvitag.rs

```diff
@@ -0,0 +1,80 @@
+//! Example to log sensor measurements of [RuuviTag]s.
+//!
+//! RuuviTags broadcast their sensor's measurements via the manufacturer
+//! specific data advertisements. Accordingly, duplicate data must be excepted.
+//!
+//! In detail this example looks for RuuviTags in reach and writes temperature,
+//! humidity and air pressure to `stdout` for each measurement.
+//!
+//!
+//! [RuuviTag]: https://ruuvi.com/ruuvitag-specs/
+
+use bluez_async::{BluetoothSession, DiscoveryFilter, BluetoothEvent, DeviceEvent};
+use futures::stream::StreamExt;
+
+/// The [Bluetooth company identifier](https://www.bluetooth.com/specifications/assigned-numbers/company-identifiers/) of Ruuvi Innovations Ltd.
+const RUUVI_ID: u16 = 0x0499;
+
+/// Protocol version of RuuviTags' data format [RAWv2](https://github.com/ruuvi/ruuvi-sensor-protocols/blob/master/dataformat_05.md)
+const PROTOCOL_VERSION: u8 = 0x05;
+
+/// Search for manufacturer data from a Ruuvi device with protocol version 5.
+fn get_ruuvi_data(mdata: impl Iterator<Item = (u16, Vec<u8>)>) -> Option<Vec<u8>> {
```

Comment:
Why not just take a `&HashMap<...>` and look up the key as usual?

Reply:
It's just a habit of me to take `Iterators` whenever possible as this decouples producer and consumer and as I rarely expect the `HashMap` to have more than 1 or 2 entries I don't mind the overhead (if there is any).
However, for the sake of simplicity of an example I can change this of course.